src/rules/preferConstRule.ts

```diff
@@ -0,0 +1,219 @@
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import * as ts from "typescript";
+import * as Lint from "../index";
+import {isNodeFlagSet} from "../language/utils";
+
+export class Rule extends Lint.Rules.AbstractRule {
+    /* tslint:disable:object-literal-sort-keys */
+    public static metadata: Lint.IRuleMetadata = {
+        ruleName: "prefer-const",
+        description: "Requires that variable declarations use `const` instead of `let` if possible.",
+        descriptionDetails: Lint.Utils.dedent`
+            If a variable is only assigned to once when it is declared, it should be declared using 'const'`,
+        optionsDescription: "Not configurable.",
+        options: null,
+        optionExamples: ["true"],
+        type: "maintainability",
+        typescriptOnly: false,
+    };
+    /* tslint:enable:object-literal-sort-keys */
+
+    public static FAILURE_STRING_FACTORY = (identifier: string) => {
+        return `Identifier '${identifier}' is never reassigned; use 'const' instead of 'let'.`;
+    }
+
+    public apply(sourceFile: ts.SourceFile): Lint.RuleFailure[] {
+        const preferConstWalker = new PreferConstWalker(sourceFile, this.getOptions());
+        return this.applyWithWalker(preferConstWalker);
+    }
+}
+
+class PreferConstWalker extends Lint.BlockScopeAwareRuleWalker<{}, ScopeInfo> {
+    public createScope() {
+        return {};
+    }
+
+    public createBlockScope() {
+        return new ScopeInfo();
+    }
+
+    public onBlockScopeEnd() {
+        const seenLetStatements: { [startPosition: string]: boolean } = {};
+        for (const usage of this.getCurrentBlockScope().getConstCandiates()) {
+            let fix: Lint.Fix;
+            if (!usage.reassignedSibling && !seenLetStatements[usage.letStatement.getStart().toString()]) {
+                // only fix if all variables in the `let` statement can use `const`
+                const replacement = new Lint.Replacement(usage.letStatement.getStart(), "let".length, "const");
+                fix = new Lint.Fix(Rule.metadata.ruleName, [replacement]);
+                seenLetStatements[usage.letStatement.getStart().toString()] = true;
+            }
+            this.addFailure(this.createFailure(
+                usage.identifier.getStart(),
+                usage.identifier.getWidth(),
+                Rule.FAILURE_STRING_FACTORY(usage.identifier.text),
+                fix,
+            ));
+        }
+    }
+
+    protected visitBinaryExpression(node: ts.BinaryExpression) {
+        if (isAssignmentOperator(node.operatorToken)) {
+            this.handleLHSExpression(node.left);
+        }
+        super.visitBinaryExpression(node);
+    }
+
+    protected visitEndOfFileToken(node: ts.Node) {
+        this.onBlockScopeEnd();
```

Comment:
I originally made that change, but it caused a compilation error for a project which generated d.ts files:

```
import {a, A} from './a';

export class B {
  static thing = a;
}
```

In this code, `A` was flagged by TSLint as unnecessary to export, but since `a` is a var and `A` is the type of `a`, it needed to import `A` also.